### DISCUSSION TITLE
what is good for  memory

### DISCUSSION THREAD

**Discussion by unkownerror1** [openzfs/zfs#14635] [Q&A]
I inquired about relevant information, 1TB of data is configured with 1G of memory, does it require at least 1G or 1G is the best , and how to draw the conclusion that 1TB is configured with 1G of memory?

**Comment by amotin** [comment]
It is a very approximate recommendation.  Requirements becomes much more critical when you use dedup, or have large L2ARC. Otherwise you should manage with as low as 8GB.  But the more RAM available -- the more will be read caching and prefetching, bigger transaction groups (more write caching), faster will be scrubs, etc.

**Reply by unkownerror1** [reply]
If dedup and l2arc are not used, regardless of the performance, whether the amount of data is large or small, is the memory requirement not high?

**Reply by amotin** [reply]
I am not exactly getting your last question.  Amount of RAM required depends on many other factors.  For example, small amount of RAM may limit how big transaction group can grow, that may limit throughput, increase write overhead and the dirty data will also eat out of ARC size.  Below 4GB ZFS automatically disables prefetch, that will affect performance.  Etc.

**Reply by unkownerror1** [reply]
I don’t know if my understanding is correct,small amount of RAM may affect the performance, but zfs can still run normally

**Reply by amotin** [reply]
Sure there still can be a dragons, same as in any rarely tested scenario, but everything is supposed to work.  Just on weekend I tried for fun to run ZFS on 32bit FreeBSD VM and it was working fine, at least on basic level, being limited to only ~1GB of KVA.
